### PR TITLE
[CI] Fix maven install failure caused by spotbugs check

### PR DESCRIPTION
**Pull request by BewareMyPower**
### Motivation

[#8784](https://github.com/apache/pulsar/pull/8784) introduced spotbugs plugin for pulsar-package-management module and its submodules. However, [#8744](https://github.com/apache/pulsar/pull/8744) cannot pass the spotsbug check but all CI passed because it's an earlier PR that didn't include the spotbugs commit.

Therefore, all PRs that rebased to the latest commit fail with all the CI checks. This PR is to fix it.

### Modifications

Remove the private `DLOutputStream#writeAsync` method that is never used.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

### CODE REVIEW

**Comment by BewareMyPower** [comment]
By the way, for pulsar committers, I suggested that if a PR contains changes **that may affect the basic build workflow**, it should be reviewed carefully. After it's merged, the other PRs should be checked if they rebased to the latest commit. If not, we should remind authors to rebase to the latest commit.

If I remember correctly, the similar problem happened not long ago. And there're a lot of spotbugs plugins to be added to each module. If PRs were not reviewed carefully, the problem would be endless.

**Review by merlimat — APPROVED** [review]

**Pull request merged**
